include/directinput/Manager.h

```diff
@@ -28,9 +28,12 @@ namespace Framework
 
 			void					CreateKeyboard();
 			void					CreateJoysticks();
+			std::list<GUID>			GetDevicesIds();
 			bool					GetDeviceInfo(const GUID&, DIDEVICEINSTANCE*);
 			bool					GetDeviceObjectInfo(const GUID&, uint32, DIDEVICEOBJECTINSTANCE*);
 
+			void					SetVibration(GUID deviceId, uint8 largeMotor, uint8 smallMotor);

```

Comment:
I think I would rename this to `SetJoystickVibration` since this only works on joysticks.